immuni_analytics/helpers/safety_net.py

```diff
@@ -97,23 +105,69 @@ def _get_jws_payload(jws_token: str) -> Dict[str, Any]:
     :param jws_token: the jws token to get the payload from.
     :return: the jws token payload.
     """
-    return _parse_jws_part(_get_jws_part(jws_token, 1))
+    try:
+        return _parse_jws_part(_get_jws_part(jws_token, 1))
+    except (JSONDecodeError, binascii.Error, IndexError, ValueError) as exc:
+        _LOGGER.warning(
+            "Could not retrieve payload from jws token.",
+            extra=dict(error=str(exc), jws_token=jws_token),
+        )
+        raise SafetyNetVerificationError()
+
+
+def _get_certificates(header: Dict[str, Any]) -> List[bytes]:
+    try:
+        certificates_string = header["x5c"]
+    except KeyError:
+        _LOGGER.warning(
+            "Could not retrieve certificates from jws header.",
+            extra=dict(header=header),
+        )
+        raise SafetyNetVerificationError()
+
+    try:
+        certificates = [base64.b64decode(c) for c in certificates_string]
+    except binascii.Error as exc:
+        _LOGGER.warning(
+            "Could not decode jws header certificates.",
+            extra=dict(error=str(exc), certificates_string=certificates_string),
+        )
+        raise SafetyNetVerificationError()
+
+    return certificates
+
+
+def _get_leaf_certificate_x509(leaf_certificate_bytes: bytes):
+    try:
+        return crypto.load_certificate(crypto.FILETYPE_ASN1, leaf_certificate_bytes)
+    except certvalidator.errors.ValidationError as exc:
+        _LOGGER.warning(
+            "Could not load the leaf certificate.",
+            extra=dict(error=str(exc), leaf_certificate_bytes=leaf_certificate_bytes),
+        )
+        raise SafetyNetVerificationError()
 
 
-def _verify_certificates(certificates: List) -> crypto.X509:
+def _validate_certificates(
+        leaf_certificate_x509: crypto.X509,
+        other_certificates_bytes: List[bytes]
```

Comment:
Not a fan of the Hungarian notation, we never used it in the whole codebase so I would avoid. What do you think?

Reply:
Fair enough, there's typing so it shouldn't be a concern!

Reply:
Fixed ✅